backend/src/services/folder-commit/folder-commit-service.ts

```diff
@@ -0,0 +1,1664 @@
+}
+
+export enum CommitType {
+  ADD = "add",
+  DELETE = "delete"
+}
+
+enum ResourceType {
+  SECRET = "secret",
+  FOLDER = "folder"
+}
+
+type TCreateCommitDTO = {
+  actor: {
+    type: string;
+    metadata?: {
+      name?: string;
+      id?: string;
+    };
+  };
+  message?: string;
+  folderId: string;
+  changes: {
+    type: string;
+    secretVersionId?: string;
+    folderVersionId?: string;
+    isUpdate?: boolean;
+  }[];
+};
+
+type TCommitChangeDTO = {
+  folderCommitId: string;
+  changeType: string;
+  secretVersionId?: string;
+  folderVersionId?: string;
+};
+
+export type ResourceChange = {
+  type: string;
+  id: string;
+  versionId: string;
+  oldVersionId?: string;
+  changeType: ChangeType;
+  commitId: bigint;
+  createdAt?: Date;
+  parentId?: string;
+  isUpdate?: boolean;
+  secretKey?: string;
+  secretVersion?: string;
+  secretId?: string;
+  folderName?: string;
+  folderVersion?: string;
+  fromVersion?: string;
+  versions?: {
+    secretKey?: string;
+    secretComment?: string;
+    skipMultilineEncoding?: boolean | null;
+    secretReminderRepeatDays?: number | null;
+    secretReminderNote?: string | null;
+    metadata?: unknown;
+    tags?: string[] | null;
+    secretReminderRecipients?: string[] | null;
+    secretValue?: string;
+    name?: string;
+  }[];
+};
+
+type ActorInfo = {
+  actorType: string;
+  actorId?: string;
+  message?: string;
+};
+
+type StateChangeResult = {
+  secretChangesCount: number;
+  folderChangesCount: number;
+  totalChanges: number;
+};
+
+type TFolderCommitServiceFactoryDep = {
+  folderCommitDAL: TFolderCommitDALFactory;
+  folderCommitChangesDAL: TFolderCommitChangesDALFactory;
+  folderCheckpointDAL: TFolderCheckpointDALFactory;
+  folderCheckpointResourcesDAL: TFolderCheckpointResourcesDALFactory;
+  folderTreeCheckpointDAL: TFolderTreeCheckpointDALFactory;
+  folderTreeCheckpointResourcesDAL: TFolderTreeCheckpointResourcesDALFactory;
+  userDAL: TUserDALFactory;
+  identityDAL: TIdentityDALFactory;
+  folderDAL: TSecretFolderDALFactory;
+  folderVersionDAL: TSecretFolderVersionDALFactory;
+  secretVersionV2BridgeDAL: TSecretVersionV2DALFactory;
+  secretV2BridgeDAL: secretV2BridgeDal.TSecretV2BridgeDALFactory;
+  projectDAL: Pick<TProjectDALFactory, "findById">;
+  folderCommitQueueService?: Pick<
+    TFolderCommitQueueServiceFactory,
+    "scheduleTreeCheckpoint" | "createFolderTreeCheckpoint"
+  >;
+  permissionService?: TPermissionServiceFactory;
+};
+
+export const folderCommitServiceFactory = ({
+  folderCommitDAL,
+  folderCommitChangesDAL,
+  folderCheckpointDAL,
+  folderTreeCheckpointDAL,
+  folderCheckpointResourcesDAL,
+  userDAL,
+  identityDAL,
+  folderDAL,
+  folderVersionDAL,
+  secretVersionV2BridgeDAL,
+  projectDAL,
+  secretV2BridgeDAL,
+  folderTreeCheckpointResourcesDAL,
+  folderCommitQueueService,
+  permissionService
+}: TFolderCommitServiceFactoryDep) => {
+  const appCfg = getConfig();
+
+  const checkProjectPermission = async ({
```

Comment:
let's update the name to be more descriptive - as in like it's for checking read commits permission